wikimedia/wikipedia-ios
# Add Native Source Editor - Part 1

**Phabricator:** 
https://phabricator.wikimedia.org/T331936

### Notes
This PR adds the native editor feature flag and implements the initial pieces of it within `PageEditorViewController`. I also added a full source edit button in `ArticleViewController`, so that we can test performance against a large set of wikitext as we build this. The entry point of this full source edit button is likely to change before release, though.

Note I found a bug in the default automatic theme switching in this branch. Marking as draft while I look into that.

### Test Steps
1. In the Staging scheme, load the article editor, either by tapping the full source "Edit" button at the top of an article or any edit pencil. 
2. Text formatting toolbars and input views should display like our Code Mirror editor, though state logic in the Find and Replace toolbar view will act wrong. I will fix this in a separate find & replace PR later on.
